### PR TITLE
flowlogs-pipeline integration

### PR DESCRIPTION
**Pull request by jotak**
Replacing the container with flowlogs-pipeline:latest, and adapting the
ConfigMap. Also it needs get/list/watch nodes.

Lot of renaming.

Needs plugin PR: https://github.com/netobserv/network-observability-console-plugin/pull/78

### CODE REVIEW

**Comment by jotak** [comment]
For the timestamp issue, I can see that `TimeFlowStart` / `End` are 0. I remember that there was an issue like that previously in goflow-kube, too. Maybe just bumping goflow2 would fix it (fl2m is one version behind goflow-kube on that)

**Review by eranra — APPROVED** [review]
Based on additional changes in https://github.com/jotak/network-observability-operator/pull/1/files

File: controllers/goflowkube/goflowkube_objects.go
Comment (eranra) [inline controllers/goflowkube/goflowkube_objects.go:29]: I added https://github.com/netobserv/flowlogs2metrics/issues/86 to close this gap ... once this is completed we will be able to return those checks.

File: controllers/goflowkube/goflowkube_objects.go
Comment (eranra) [inline controllers/goflowkube/goflowkube_objects.go:55]: @jotak this works ... It is possible to use the struct from here:  https://github.com/netobserv/flowlogs2metrics/blob/051a608b1368f25986bcfb1880b416e7d57d943a/pkg/api/ingest_collector.go#L20 directly instead of reimplementing the API struct. The value is that we can be sure that we are aligned to some version of flowlogs2metrics.

Comment (jotak) [reply controllers/goflowkube/goflowkube_objects.go:55]: Yes, but until now we've tried to keep the operator loosely coupled with other models to avoid dependency complexity and keep vendors light. We can revisit this choice if we feel it's useful.

**Comment by jotak** [comment]
It's ready for review, along with https://github.com/netobserv/network-observability-console-plugin/pull/78
I keep it on hold in order to do a release before merging, as discussed with @memodi 

/hold

File: controllers/flowlogspipeline/flp_objects.go
Comment (jpinsonneau) [inline controllers/flowlogspipeline/flp_objects.go:281]: Should we expose `decode` `transform` & `write` in `FlowCollector` config if you simply want to collect and send to something else without using the console UI ?

**Review by jpinsonneau — APPROVED** [review]

File: controllers/flowlogspipeline/flp_objects.go
Comment (jotak) [reply controllers/flowlogspipeline/flp_objects.go:281]: Yes that could be interesting. For now I prefer to focus on FLP integration with as few changes as possible in the current API, but we can expect more config exposition soon. We need also to draw the line between something opinionated or not. FLP is flexible, but the operator could be more opinionated (which means easier to configure). Not saying it has to be.. it's a discussion to have

File: controllers/flowlogspipeline/flp_objects.go
Comment (mariomac) [inline controllers/flowlogspipeline/flp_objects.go:55]: This structure with change after this PR: https://github.com/netobserv/flowlogs-pipeline/pull/83

Anyway, I think this structure will remain constant. Am I right? In that case, for simplification, we can just bundle the configuration as a YAML text/file and embed it into the FLP Pod descriptor.

**Review by mariomac — APPROVED** [review]

File: controllers/flowlogspipeline/flp_objects.go
Comment (jotak) [reply controllers/flowlogspipeline/flp_objects.go:55]: It's not entirely constant, in particular the Loki part is exposed through CR

File: controllers/flowlogspipeline/flp_objects.go
Comment (jotak) [reply controllers/flowlogspipeline/flp_objects.go]: that fine I've figured out

**Pull request merged**
